### ISSUE TITLE
The splunk::forwarder::purge_inputs generates error 

### ISSUE THREAD

**Issue by Vincent--** [voxpupuli/puppet-splunk#69]
### Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 3.8.7
- Ruby: ruby 1.9.3p484 (2013-11-22 revision 43786) [x86_64-linux]
- Distribution: Linux siem 4.4.0-31-generic #50~14.04.1-Ubuntu SMP Wed Jul 13 01:07:32 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
- Module version: 5.0.1

### How to reproduce (e.g Puppet code you use)
Try to enable splunk::forwarder::purge_inputs

```
splunk::forwarder::purge_inputs     : true
```

### What are you seeing

An error message is generated during the provision:
```
Notice: Compiled catalog for puppet in environment production in 4.92 seconds
Error: /Stage[main]/Splunk::Params/Splunk_config[splunk]: Failed to generate additional resources using 'generate': can't convert nil into String
Notice: Finished catalog run in 15.65 seconds
```

### What behaviour did you expect instead
No error message

**Comment by callafeucht** [comment]
I am also encountering this error.

- Puppet: 3.7.5
- Ruby: ruby 2.0.0p598 (2014-11-13) [x86_64-linux]
- Distribution: CentOS Linux release 7.2.1511 (Core) 
- Module version: v5.0.2

**Comment by michaelburgener** [comment]
I'm having the same issue as well

- Puppet: 4.8.1
- Ruby: ruby 2.1.9p490 (2016-03-30 revision 54437) [x86_64-linux]
- Distribution: Ubuntu 16.04.1 LTS Linux 4.4.0-51-generic x86_64
- Module version: v5.1.1-rc0

**Comment by wottah** [comment]
+1
Also encountered this problem. The issue seems to come from within the inifile dependency.

**Comment by TraGicCode** [comment]
This was fixed in #96 and i also tested this locally with beaker.
